.github/workflows/lint-and-test.yml

```
@@ -136,6 +136,41 @@ jobs:
       - name: Run clippy::pedantic for proof-of-sql-parser
         run: cargo clippy --lib -p proof-of-sql-parser -- -D clippy::pedantic
 
+  coverage:
+    name: Code Coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+      - name: Install Dependencies
+        run: sudo apt-get update && sudo apt-get install -y clang lld
+      - uses: taiki-e/install-action@cargo-llvm-cov
+      - name: Clean Previous Coverage Artifacts
+        run: cargo llvm-cov clean --workspace
+      - name: Run Tests to Generate Coverage Data (All Features)
+        run: cargo llvm-cov --no-report --all-features
+      - name: Run Tests to Generate Coverage Data (Rayon Only)
+        run: cargo llvm-cov --no-report --no-default-features --features="rayon"
+      - name: Run Tests to Generate Coverage Data (Blitzar Only)
+        run: cargo llvm-cov --no-report --no-default-features --features="blitzar"
+      - name: Run Tests to Generate Coverage Data (std only)
+        run: cargo llvm-cov --no-report --no-default-features --features="std"
```

A lot of these tests are redundant, and we should be more selective in order to be more efficient.

I'm happy to merge this as is and mark the issue complete, with the caveat that since the job takes 51m to run, let's comment these lines out. I want another issue created for identifying long-running tests as well as redundant test runs due to feature flags.

```suggestion
      #- name: Run Tests to Generate Coverage Data (Rayon Only)
      #  run: cargo llvm-cov --no-report --no-default-features --features="rayon"
      #- name: Run Tests to Generate Coverage Data (Blitzar Only)
      #  run: cargo llvm-cov --no-report --no-default-features --features="blitzar"
      #- name: Run Tests to Generate Coverage Data (std only)
      #  run: cargo llvm-cov --no-report --no-default-features --features="std"
```